docs/reference/modules/indices/search-settings.asciidoc

```diff
@@ -24,4 +24,11 @@ few resources.
 Maximum number of <<search-aggregations-bucket,aggregation buckets>> allowed in
 a single response. Defaults to 65,535.
 +
-Requests that attempt to return more than this limit will return an error.
\ No newline at end of file
+Requests that attempt to return more than this limit will return an error.
+
+[[indices-query-bool-max-nested-depth]]
+`indices.query.bool.max_nested_depth`::
+(<<static-cluster-setting,Static>>, integer) Maximum nested depth of bool queries. Defaults to `20`.
++
+This setting limits the nesting depth of bool queries. Nesting too deep will lead to stack overflow
+errors.
```

Comment:
May be a little paraphrase the last sentence: "Deep nesting of boolean queries may lead to stack overflow."